src/fs/parseSource.ts

```diff
@@ -108,6 +110,10 @@ export async function main(
 
   (await fileEntry(source, ignores)).forEach((file) => {
     try {
+      if (isBinary(file)) {
+        onBinaryFile && onBinaryFile(file);
```

Comment:
question: Would it be better to assert the function is defined? This feels like an error state.

Reply:
Originally I had required the parameter but I made it optional since not every client will care (for example, when generating snippets, we just ignore binary files). I can definitely make it required so a client is always explicitly checking binary files, even if they just discard them. Thoughts?

Reply:
I would prefer it to be optional, but don't feel strongly about it.